### DISCUSSION TITLE
Stabilize interpolation

### DISCUSSION THREAD

**Discussion by vudongpham** [davidfrantz/force#306] [Ideas]
Sometime when you fit the interpolation function with FORCE, there are some unmeaningful spectral values (e.g., negative or over 10000) when the temporal data is sparse, which happens a lot when go annually. Test case:

TILTE = X0068_Y0043
SENSORS = LND08 LND07
ABOVE_NOISE = 0
BELOW_NOISE = 0
DATE_RANGE = 2014-01-01 2014-12-31
DOY_RANGE = 1 365
INDEX = NIR
INTERPOLATE = HARMONIC
HARMONIC_MODES = 3
HARMONIC_FIT_RANGE = 2014-01-01 2014-12-31
INT_DAY = 10

Examples from some seleced pixels:

![original](https://github.com/davidfrantz/force/assets/123673251/973a551e-de46-48f4-aa9e-b16ed900e4f2)

So here is my idea, we can dublicate the time-series information back and forth and then fit the interpolation function to the entire data range, then cut the middle for the final results. This way, the interpolated data is much more stable. Bellow is an example:

![stablized](https://github.com/davidfrantz/force/assets/123673251/86f96bad-f2a5-4f16-b667-dbf36d3318eb)

And here is when comparing with FORCE:

![compare](https://github.com/davidfrantz/force/assets/123673251/54320ebc-4a6e-4d90-86ba-ddc8f1c53f6f)

**Comment by davidfrantz** [comment]
Hey Dong,

I guess this is because this type of interpolation should rather be used with multi-annual data. Your solution is pretty logical, but I am not sure if it makes sense to implement such a special case. If you would reduce the time window even further, other things might happen, too.

Happy to hear other user's thoughts, though

Cheers,
David

**Reply by vudongpham** [reply]
Hi David,
The linear interpolation did not seem to have this effect and I have not tried with other methods. I implemented this in Python (not UDF) and it took arround 2 mins to interpolate one tile (1000x1000) from TSS data using 8 cores cpu with paralell. 
I saw many used annual harmonic features for classification workflows, so I think this might helps. Happy to year others' thoughts as well!

Best,
Dong
